lib/erb.rb

```diff
@@ -885,8 +885,16 @@ def run(b=new_toplevel)
   #
   # _b_ accepts a Binding object which is used to set the context of
   # code evaluation.
+  # _locals_ accepts a Hash object which is used to set local variables
+  # on the Binding object.
   #
-  def result(b=new_toplevel)
+  def result(b=new_toplevel, locals: {})
+    unless locals.empty?
```

Comment:
> I think the default should be nil and if locals.

I agreed because it may be faster. Fixed so https://github.com/ruby/ruby/pull/1618/commits/c39a419741cc2e3afe9f78bd2b3f9aa5ac85eb77.